src/layouts/fancy-guide.tsx

```diff
@@ -17,14 +18,20 @@ const FancyGuideLayout: FunctionComponent<LayoutProps> = ({
   children,
   meta = {},
 }) => {
+  const router = useRouter()
+  const url = process.env.NEXT_PUBLIC_DEPLOYMENT_URL + router.asPath
+
   const {
     title,
     description,
     titleAppendSiteName = false,
-    url,
+    url: urlFromProps,
     ogImage,
     contributors,
   } = meta
+
+  const canonicalUrl = urlFromProps ? urlFromProps : url
```

Comment:
updates our <FancyGuide> component used on a few MDX pages to set the canonicalUrl from the router if none was passed. Currently I couldn't find any of these pages that use the component where we pass a url.